### PR TITLE
Do not update LocalQueue counters if tracing is disabled

### PR DESCRIPTION
**Pull request by vasilmkd**

### CODE REVIEW

**Review by armanbilge — COMMENTED** [review]
Apologies, I don't think I've paid enough attention: how does stack tracking enabled/disabled relate to the JMX metrics? (Are these counters JMX metrics?)

**Comment by vasilmkd** [comment]
These counters are the implementation details of the jmx metrics. Those interfaces show these numbers.

The MBeans are not registered if tracing is disabled, so there's no need to calculate them. I disabled them because there are some places where we have to publish them more strictly, so this saves a memory barrier here and there.

File: core/jvm/src/main/scala/cats/effect/unsafe/WorkerThread.scala [hunk 1]
```diff
@@ -537,9 +540,14 @@ private final class WorkerThread(
 
       // With another `WorkerThread` started, it is time to execute the blocking
       // action.
-      val result = thunk
-
-      pool.blockedWorkerThreadCounter.decrementAndGet()
+      val result =
+        try thunk // Exceptions are caught in IOFiber.
+        finally {
```
Comment (armanbilge) [inline core/jvm/src/main/scala/cats/effect/unsafe/WorkerThread.scala:545]: Why do we need finally if exceptions are caught?

**Review by armanbilge — APPROVED** [review]
Thanks for clarifying about that! :)

File: core/jvm/src/main/scala/cats/effect/unsafe/WorkerThread.scala (hunk 1 above)
Comment (vasilmkd) [reply core/jvm/src/main/scala/cats/effect/unsafe/WorkerThread.scala:545]: If an exception occurs in the `thunk`, the counter would not be decremented. This decrements it while allowing the exception to bubble up to the proper handler.

Comment (armanbilge) [reply core/jvm/src/main/scala/cats/effect/unsafe/WorkerThread.scala:545]: Ah, the catch happens in the caller, not inside of the the `thunk`. Thanks.

**Pull request merged**